File: catalog/backends/node-express/patterns/n_plus_one_mysql_route.js
const mysql = require('mysql2/promise');

// Pattern: DBM N+1 — one query per parent row (MySQL)
// Adapt: replace table/column names with domain entities
const pool = mysql.createPool(process.env.MYSQL_URL || 'mysql://demo:demo@mysql:3306/demo');

async function itemsWithDetails(req, res) {
  const [parents] = await pool.query('SELECT * FROM parents');
  const results = [];
  for (const parent of parents) {
    const [children] = await pool.query(
      'SELECT * FROM children WHERE parent_id = ?', [parent.id]
    );
    results.push({ parent, children });
  }
  res.json(results);
}

module.exports = { itemsWithDetails };

File: catalog/backends/python-django/patterns/n_plus_one_mysql_route.py
import mysql.connector
import os
from django.http import JsonResponse

# Pattern: DBM N+1 — one query per parent row (MySQL)
# Adapt: replace table/column names with domain entities
MYSQL_CONFIG = {
    "host": os.environ.get("MYSQL_HOST", "mysql"),
    "user": os.environ.get("MYSQL_USER", "demo"),
    "password": os.environ.get("MYSQL_PASSWORD", "demo"),
    "database": os.environ.get("MYSQL_DATABASE", "demo"),
}


def items_with_details(request):
    conn = mysql.connector.connect(**MYSQL_CONFIG)
    cur = conn.cursor(dictionary=True)
    cur.execute("SELECT * FROM parents")
    parents = cur.fetchall()
    results = []
    for parent in parents:
        cur.execute("SELECT * FROM children WHERE parent_id = %s", (parent["id"],))
        results.append({"parent": parent, "children": cur.fetchall()})
    cur.close()
    conn.close()
    return JsonResponse(results, safe=False)
